### ISSUE TITLE
GeoHealthCheck per-resource email notification

### ISSUE THREAD

**Issue by simboss** [geosolutions-it/cezio#2]
See description [here](https://docs.google.com/document/d/1pvTY1ROSiNspvHlOtZFZ-mokh6DFT9KWG1QIFIua7lg/edit#).

This is currently blocked waiting for approval from Simone.

**Comment by simboss** [comment]
@cezio please, refine and confirm the estimate.

**Comment by cezio** [comment]
> would be nice to have the recipient email suggested based on the email addresses already into the system

Could you elaborate a bit? Especially, from which values should suggestions come? From users in GHC? From emails already used? If emails already used, then should it be from all emails, or emails used by specific user?

**Comment by cezio** [comment]
We could use small improvement: when adding a resource, notification should be pre-populated with user's email.

**Comment by simboss** [comment]
@cezio there is a proposal which you made ([here](https://docs.google.com/document/d/1pvTY1ROSiNspvHlOtZFZ-mokh6DFT9KWG1QIFIua7lg/edit#)) and which is sufficient.

The goal is a differentiate target notification emails for the resources. Prepopulating with GHC users' emails is fine but not sufficient as most of the time I will want to send resource's notifications to multiple emails which could not belong to GHC users.

So:
- Ability to enter multiple email addresses on a per-resource base which I would want to nitify is a must have. This email are **not** required to belong to GHC users.
- Ability to suggest emails based on GHC users is nice to have.

**Comment by cezio** [comment]
> Prepopulating with GHC users' emails is fine but not sufficient as most of the time I will want to send resource's notifications to multiple emails which could not belong to GHC users.

I understand, and I don't think this conflicts with goal here. This is about sane defaults, not any equivalent functionality. My point is that when user is adding new resource to monitor, by default resource should send notifications to that user. User still can remove that email, or add another.

About email suggestions, this still doesn't tell if suggestions should use all emails used in GHC, or just ones that are related to current user.

**Comment by simboss** [comment]
> I understand, and I don't think this conflicts with goal here. This is about sane defaults, not any equivalent functionality. My point is that when user is adding new resource to monitor, by default resource should send notifications to that user. User still can remove that email, or add another.

+1

> About email suggestions, this still doesn't tell if suggestions should use all emails used in GHC, or just ones that are related to current user.

Good point, what is the lowest hanging fruit? I would guess using them all, right?

**Comment by cezio** [comment]
> Good point, what is the lowest hanging fruit? I would guess using them all, right?

Yes, the easiest way would be to use all emails, although there should be very little difference between implementations (it's a matter of constructing one query actually). But this is important detail if we want to have some isolation between different users.

**Comment by simboss** [comment]
> Yes, the easiest way would be to use all emails, although there should be very little difference between implementations (it's a matter of constructing one query actually). But this is important detail if we want to have some isolation between different users.

Yeah, I was thinking about isolation myself.

How do you see isolation coming in here? I mean, do we have a concept of organization or anything in GHC?

**Comment by cezio** [comment]
> How do you see isolation coming in here? I mean, do we have a concept of organization or anything in GHC?

At the moment there's no facility in GHC to isolate resources. There are some plans for the future: https://github.com/geopython/GeoHealthCheck/issues/18

**Comment by simboss** [comment]
Then how would you go with having some isolation between different users? Would that impact your estimate?

**Comment by cezio** [comment]
> Then how would you go with having some isolation between different users? 

 For the moment, it's just autocomplete with emails used by user.

> Would that impact your estimate?

 No, it's a minor thing. More problem is to get autocomplete with list of values working (but I'm on it, should be ready soon).

**Comment by simboss** [comment]
cool, if there is anything blocking you from respecting the estimate let me know.

We need an initial POC to move on.

Make sure this gets contributed back to GHC.

**Comment by cezio** [comment]
for reference: https://github.com/geopython/GeoHealthCheck/pull/187
